app/src/main/java/com/github/multimatum_team/multimatum/DeadlineDetailsActivity.kt

```diff
@@ -32,40 +34,52 @@ class DeadlineDetailsActivity : AppCompatActivity() {
 
     lateinit var id: DeadlineID
     private var editMode: Boolean = true
+    private val deadlineListViewModel: DeadlineListViewModel by viewModels()
 
     private lateinit var titleView: EditText
     private lateinit var dateView: TextView
     private lateinit var detailView: TextView
     private lateinit var doneButton: CheckBox
 
-    private lateinit var date: LocalDateTime
-    private lateinit var state: DeadlineState
+    // Set them on default value, waiting the fetch of the deadlines
+    private var date: LocalDateTime = LocalDateTime.of(2022, 10, 10, 10, 10)
+    private var state: DeadlineState = DeadlineState.TODO
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_deadline_details)
-
-        // Recuperate the Deadline from the intent
-        val title = intent.getStringExtra(EXTRA_TITLE)
-        id = intent.getStringExtra(EXTRA_ID) as DeadlineID
-        date = intent.getSerializableExtra(EXTRA_DATE) as LocalDateTime
-        state = intent.getSerializableExtra(EXTRA_STATE) as DeadlineState
+        var title: String
 
         // Recuperate the necessary TextView
         titleView = findViewById(R.id.deadline_details_activity_title)
         dateView = findViewById(R.id.deadline_details_activity_date)
         detailView = findViewById(R.id.deadline_details_activity_done_or_due)
         doneButton = findViewById(R.id.deadline_details_activity_set_done)
```

Comment:
As said by CodeClimate, this function should be refactored. The variables initialization could be done outside easily, as well as the DeadlineListViewModel observation below, and the shown data update (end of function).